examples/platform/silabs/BaseApplication.cpp

```diff
@@ -37,9 +37,13 @@
 #endif // QR_CODE_ENABLED
 #endif // DISPLAY_ENABLED
 
-#if CHIP_CONFIG_ENABLE_ICD_SERVER == 1
+#if defined(CHIP_CONFIG_ENABLE_ICD_SERVER) && CHIP_CONFIG_ENABLE_ICD_SERVER
```

Comment:
We should not use defined/ifdef.

Please include relevant headers and only use `#if` and `#if !`.

Reply:
done